src/main/kotlin/org/rust/lang/core/resolve/ResolveEngine.kt

```diff
@@ -134,9 +134,20 @@ object ResolveEngine {
     fun resolveMethodCallExpr(call: RsMethodCallExpr): RsNamedElement? {
         val receiverType = call.expr.type
         val name = call.identifier.text
+        var method: RsFunction? = null
 
-        return receiverType.getMethodsIn(call.project)
-            .find { it.name == name }
+        receiverType.getMethodsIn(call.project)
+            .filter { it.name == name }
+            .forEach {
+                (it.parent as? RsImplItem)?.let {impl ->
```

Comment:
Could you add a comment here explaining that if there's a conflict, we want to prefer the inherent method?